plugin/gc-app/src/main/java/io/deephaven/app/GcPoolsConsumer.java

```diff
@@ -0,0 +1,140 @@
+                        Aggregation.AggCount("Count"),
+                        Aggregation.AggSum(
+                                "UsedReclaimedTotal=UsedReclaimed",
+                                "CommittedReclaimedTotal=CommittedReclaimed"),
+                        Aggregation.AggAvg(
+                                "UsedReclaimedAvg=UsedReclaimed",
+                                "CommittedReclaimedAvg=CommittedReclaimed"),
+                        Aggregation.AggApproxPct("UsedReclaimed",
+                                PercentileOutput.of(0.5, "UsedReclaimedP_50")),
+                        Aggregation.AggApproxPct("CommittedReclaimed",
+                                PercentileOutput.of(0.5, "CommittedReclaimedP_50")),
+                        Aggregation.AggLast(
+                                "BeforeInit",
+                                "BeforeUsed",
+                                "BeforeCommitted",
+                                "BeforeMax",
+                                "AfterInit",
+                                "AfterUsed",
+                                "AfterCommitted",
+                                "AfterMax")),
+                        "Name");
+    }
+
+    private final StreamConsumer consumer;
+    private WritableChunk<Values>[] chunks;
+    private boolean isFirst;
+
+    GcPoolsConsumer(StreamConsumer consumer) {
+        this.consumer = Objects.requireNonNull(consumer);
+        // noinspection unchecked
+        chunks = StreamToTableAdapter.makeChunksForDefinition(DEFINITION, CHUNK_SIZE);
+        isFirst = true;
+    }
+
+    public synchronized void add(GcInfo info) {
+        final Map<String, MemoryUsage> afterMap = info.getMemoryUsageAfterGc();
+        for (Entry<String, MemoryUsage> e : info.getMemoryUsageBeforeGc().entrySet()) {
+            final String poolName = e.getKey();
+            final MemoryUsage before = e.getValue();
+            final MemoryUsage after = afterMap.get(poolName);
+            if (!isFirst && equals(before, after)) {
+                continue;
+            }
+            chunks[0].asWritableLongChunk().add(info.getId());
+            chunks[1].<String>asWritableObjectChunk().add(poolName.intern());
+
+            chunks[2].asWritableLongChunk().add(negativeOneToNullLong(before.getInit()));
+            chunks[3].asWritableLongChunk().add(before.getUsed());
+            chunks[4].asWritableLongChunk().add(before.getCommitted());
+            chunks[5].asWritableLongChunk().add(negativeOneToNullLong(before.getMax()));
+
+            chunks[6].asWritableLongChunk().add(negativeOneToNullLong(after.getInit()));
+            chunks[7].asWritableLongChunk().add(after.getUsed());
+            chunks[8].asWritableLongChunk().add(after.getCommitted());
+            chunks[9].asWritableLongChunk().add(negativeOneToNullLong(after.getMax()));
+        }
+        isFirst = false;
+        if (chunks[0].size() + POOL_SIZE_UPPER_BOUND >= CHUNK_SIZE) {
```

Comment:
We should do a capacity check *before* filling these chunks. If needed size is bigger than your available capacity, `flushInternal`. If needed size is bigger than your chunk size, grow your chunk size.